### PR TITLE
Slider enhancements min

### PR DESCRIPTION
**Pull request by maxpeterson**
Based on #2 without changing whitespace and excluding the `event.stopImmediatePropagation();` that breaks IE9

### CODE REVIEW

**Comment by maxpeterson** [comment]
@grahamgilchrist can you review / merge this

**Comment by maxpeterson** [comment]
@henrahmagix do you want to look at this? It is basically #2 simplified and without the additional `event.stopImmediatePropagation();`

**Comment by henrahmagix** [comment]
I think you should include `event.preventDefault()` and `event.stopPropagation()` (both supported in IE9), for the intended effect that was part of a bug fix of #2.

File: build/angular-slider.js [hunk 1]
```diff
@@ -943,10 +954,10 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
 
                                 // save the various dimensions we'll need
                                 barWidth = width(refs.fullBar);
-                                pointerHalfWidth = halfWidth(refs.minPtr);
+                                pointerHalpointerHalfWidth = parseInt(minPtrWidth / 2);
```
Comment (henrahmagix) [inline build/angular-slider.js]: Always set the base: `parseInt(minPtrWidth / 2, 10)`

Comment (henrahmagix) [inline build/angular-slider.js]: Personally I think `minPtrWidth` should be renamed to `minPointerWidth` so it can be related to `pointerHalfWidth`.

Comment (henrahmagix) [inline build/angular-slider.js]: typo: `pointerHalpointerHalfWidth` -> `pointerHalfWidth`

File: build/angular-slider.js [hunk 2]
```diff
@@ -1146,8 +1157,12 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
 
                                     // set the low knob's and bubble's new positions
                                     offset(refs.minPtr, offsetFromPercent(stretchedLowPercent));
-                                    offset(refs.lowBub,
-                                        offsetFromPercent(percentFromOffset(offsetLeft(refs.minPtr) - halfWidth(refs.lowBub) + pointerHalfWidth)));
+                                    var bubbleLeftOffset = offsetFromPercent(percentFromOffset(offsetLeft(refs.minPtr) - halfWidth(refs.lowBub) + pointerHalfWidth));
+                                    // If the pointer width has been overridden, ensure that the bubble is placed at the same position as the pointer
+                                    if(scope.pointerWidthOverride !== undefined) {
```
Comment (henrahmagix) [inline build/angular-slider.js]: Space after `if`

File: build/angular-slider.js [hunk 3]
```diff
@@ -1395,7 +1410,10 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
                                  */
                                 function redrawSliders() {
                                     setPointers();
-                                    adjustBubbles();
+                                    // Only adjust bubbles if allowed
+                                    if(scope.noBubbleAdjustment === undefined) {
```
Comment (henrahmagix) [inline build/angular-slider.js]: Space after `if` (it was greg's style which is inconsistent with our style guide)

File: build/angular-slider.js [hunk 4]
```diff
@@ -1440,6 +1458,23 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
...
                                             var currentX = event.clientX || event.x;
 
+                                            /**
+                                             * Because event.clientX and event.x are not included in the touch event object because of the potential
+                                             * for multi-touch, this information is obtained from the first touch event element in the touch event 
+                                             * array.
+                                             * If this array is empty, the function is returned. This results in the last value being used as opposed 
+                                             * to the value resetting to 0.
+                                             */
+                                            if(currentX == undefined) {
```
Comment (henrahmagix) [inline build/angular-slider.js]: Space after `if`

File: build/angular-slider.js [hunk 5]
```diff
@@ -1440,6 +1458,23 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
...
+                                                if(event['touches'] !== undefined && event['touches'].length) {
```
Comment (henrahmagix) [inline build/angular-slider.js]: Space after `if`

File: build/angular-slider.js [hunk 6]
```diff
@@ -1440,6 +1458,23 @@ angular.module('vr.directives.slider', ['ngTouch']).directive('slider',
...
+                                                    currentX = event.touches[0].clientX;
+                                                } else if(event['originalEvent'] !== undefined && event['originalEvent']['touches'] !== undefined && event['originalEvent']['touches'].length) {
```
Comment (henrahmagix) [inline build/angular-slider.js]: Space after `if`

File: build/angular-slider.js (hunk 5 above)
Comment (henrahmagix) [inline build/angular-slider.js]: Use dot notation to access properties of `event` in this whole conditional block

File: build/angular-slider.js (hunk 6 above)
Comment (henrahmagix) [inline build/angular-slider.js]: Actually ignore: I see the file already has `if()`

**Comment by henrahmagix** [comment]
hah woops, I've commented on the build file, but it's the same as the source.

File: Gruntfile.js [hunk 7]
```diff
@@ -20,12 +20,24 @@ module.exports = function(grunt){
                 src: ['<%= srcFiles %>'],
                 dest: 'build/<%= pkg.name %>.min.js'
             }
-        }
+        },
+        watch: {
+            options: {
```
Comment (grahamgilchrist) [inline Gruntfile.js]: You probably don't need the `livereload` property here?

Comment (henrahmagix) [inline Gruntfile.js]: I agree, it instead should be set by the project.

Comment (henrahmagix) [inline Gruntfile.js]: Oh wait, this is the project, really. I think it's fine, if it works.

File: build/angular-slider.js (hunk 3 above)
Comment (grahamgilchrist) [inline build/angular-slider.js]: If we are going to submit this fork back, we should theoretically be following the style guide of the existing code.

**Comment by grahamgilchrist** [comment]
I'm not sure about adding `event.stopPropagation()` by default, as this can potentially break intended functionality on other parts of the page? At the least I think it ought to be a config option

**Comment by henrahmagix** [comment]
@grahamgilchrist `event.stopImmediatePropagation()` was included in #2 for some reason, but it broke IE9, so the similar `event.stopPropagation()` should be included here for the same fix. What that fix is, or whether `event.stopImmediatePropagation()` was needed in the first place, I'm not sure.

**Comment by grahamgilchrist** [comment]
@henrahmagix I know it was in #2, but without an issue or commit message we don't know why. 
I don't think library style widgets should be doing `stopPropagation()` by default without an option, because it can cause breakage in unrelated other parts of the application relying on certain events. It makes the components less isolated.

`e.preventDefault()` is less of an issue as it has predictable behaviour only on the element specified.

**Comment by henrahmagix** [comment]
:+

**Comment by grahamgilchrist** [comment]
Perhaps better to leave out until we discover a need for that fix again, and then we can add a properly commented code and PR to reinstate an option `stopPropagation()`

File: build/angular-slider.js (hunk 5 above)
Comment (maxpeterson) [inline build/angular-slider.js]: Greg changed these to be like this in a subsequent commit so I was assuming there was good reason (I must confess I am not sure why)

File: build/angular-slider.js (hunk 3 above)
Comment (maxpeterson) [inline build/angular-slider.js]: @henrahmagix it is not greg's library

Comment (henrahmagix) [inline build/angular-slider.js]: Yes, but I also know that Greg preferred writing `if()` when he was at Incuna, even though Incuna's style guide says `if ()` =)

**Comment by maxpeterson** [comment]
@henrahmagix @grahamgilchrist updated. I don;t think we should restore `stopPropagation()` until we know what it was required for I also would not consider it a bug fix. It is potentially a breaking changes.

**Pull request merged**